solutions/CF They are Everywhere.mdx

```diff
@@ -0,0 +1,181 @@
+---
+id: cf-701C
+source: CF
+title: They are Everywhere
+author: Kevin Sheng
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/46283)
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <iostream>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+ 
+using std::cout;
+using std::endl;
+using std::vector;
+ 
+/**
+ * https://codeforces.com/problemset/problem/701/C
+ * 7
+ * bcAAcbc should output 3
+ * 6
+ * aaBCCe should output 5
+ */
+int main() {
+    int flat_num;
+    std::cin >> flat_num;
+    vector<char> flats(flat_num);
+    std::unordered_set<char> types;
+    for (char& p : flats) {
+        std::cin >> p;
+        types.insert(p);
+    }
+ 
+    int lo = 0;
+    int hi = flat_num;
+    int valid = -1;
+    while (lo <= hi) {
+        int mid = (lo + hi) / 2;
+ 
+        bool all_catchable = false;    
+        std::unordered_map<char, int> curr_pokemon;
```

Comment:
You can use array of len 128 instead (chars are from -128 to 127, but in practice they shouldn't be negative)

Reply:
Not really necessary. As long as it consistently gets AC'd (which it should), it should be fine.

Reply:
ok